src/runtime/recipe/handle-connection.ts

```diff
@@ -77,15 +76,22 @@ export class HandleConnection {
   getQualifiedName() { return `${this.particle.name}::${this.name}`; }
   get tags() { return this._tags; }
   get type() {
-    if (this._type) {
-      return this._type;
+    if (this._rawType) {
+      return this._rawType;
     }
-    return this._rawType;
+    const spec = this.spec;
+    return spec ? spec.type : null;
   }
   get rawType() {
```

Comment:
we probably don't need this any more either.

Reply:
removed